client/post-editor/post-editor.jsx

```diff
@@ -1013,17 +1014,268 @@ export const PostEditor = React.createClass( {
+
+		if ( lastLtPos > lastGtPos ) {
+			// inside a tag that was opened, but not closed
+
+			// find what the tag is
+			const tagContent = content.substr( lastLtPos );
+			const tagMatch = tagContent.match( /<\s*(\/)?(\w+)/ );
+			if ( ! tagMatch ) {
+				return null;
+			}
+
+			const tagType = tagMatch[ 2 ];
+			const closingGt = tagContent.indexOf( '>' );
+			const isClosingTag = ! ! tagMatch[ 1 ];
+
+			return {
+				ltPos: lastLtPos,
+				gtPos: lastLtPos + closingGt + 1, // offset by one to get the position _after_ the character,
+				tagType,
+				isClosingTag,
+			};
+		}
+		return null;
+	},
+
+	addHTMLBookmarkInTextAreaContent: function() {
+		const textArea = this.editor._editor.getElement();
+
+		let htmlModeCursorStartPosition = textArea.selectionStart;
+		let htmlModeCursorEndPosition = textArea.selectionEnd;
+
+		// check if the cursor is in a tag and if so, adjust it
+		const isCursorStartInTag = this.isCursorInATag( textArea.value, htmlModeCursorStartPosition );
+		if ( isCursorStartInTag ) {
+			htmlModeCursorStartPosition = isCursorStartInTag.ltPos;
+		}
+
+		const isCursorEndInTag = this.isCursorInATag( textArea.value, htmlModeCursorEndPosition );
+		if ( isCursorEndInTag ) {
+			htmlModeCursorEndPosition = isCursorEndInTag.gtPos;
+		}
+
+		let mode = 'single';
+		if ( htmlModeCursorStartPosition !== htmlModeCursorEndPosition ) {
+			mode = 'range';
+		}
+
+		const bookMarkStart = '<span ' +
+			'data-mce-type="bookmark"	' +
+			'id="mce_SELREST_start" ' +
+			'data-mce-style="overflow:hidden;line-height:0px" ' +
+			'style="overflow:hidden;line-height:0px"' +
+			'>' +
+			'&#65279;' +
+			'</span>';
+
+		let bookMarkEnd = null;
+		if ( mode === 'range' ) {
+			bookMarkEnd = '<span ' +
+				'data-mce-type="bookmark"	' +
+				'id="mce_SELREST_end" ' +
+				'data-mce-style="overflow:hidden;line-height:0px" ' +
+				'style="overflow:hidden;line-height:0px"' +
+				'>' +
+				'&#65279;' +
+				'</span>';
+
+			// // TODO detect if you're not in a tag to not break things
+			textArea.value = [
+				textArea.value.slice( 0, htmlModeCursorStartPosition ),
+				bookMarkStart,
+				textArea.value.slice( htmlModeCursorStartPosition, htmlModeCursorEndPosition ),
+				bookMarkEnd,
+				textArea.value.slice( htmlModeCursorEndPosition ),
+			].join( '' );
+		} else {
+			textArea.value = [
+				textArea.value.slice( 0, htmlModeCursorStartPosition ),
+				bookMarkStart,
+				textArea.value.slice( htmlModeCursorStartPosition )
+			].join( '' );
+		}
+
+		this.editor.onTextAreaChange( { target: { value: textArea.value } } );
+	},
+
+	focusHTMLBookmarkInVisualEditor: function( ed ) {
+		const startNode = ed.target.getDoc().getElementById( 'mce_SELREST_start' );
+		const endNode = ed.target.getDoc().getElementById( 'mce_SELREST_end' );
+
+		if ( ! startNode ) {
+			return;
+		}
+
+		ed.target.focus();
+
+		if ( ! endNode ) {
+			ed.target.selection.select( startNode );
+		} else {
+			const sel = document.createRange();
```

Comment:
Is this an abbreviation of `selection`? Are we avoiding the whole word to avoid variable shadowing or something similar?

Reply:
More likely leftover of previous version of the code where both `sel` and `selection` existed. Updated to be more descriptive.